File: Python/5 kyu/Pick peaks/main.py
def goesLower(arr, i):
    for k in range(i, len(arr)):
        if arr[k] < arr[i]:
            return True
        elif arr[k] > arr[i]:
            return False
    return False

def pick_peaks(arr):
    pos = []
    peaks = []
    for ind in range(1, len(arr) - 1):
        if arr[ind] > arr [ind - 1] and arr[ind] >= arr[ind + 1] and goesLower(arr, ind):
            pos.append(ind)
            peaks.append(arr[ind])
    return {"pos": pos, "peaks": peaks}

if __name__ == "__main__":
    print(pick_peaks([3,2,3,6,4,1,2,3,2,1,2,2,2,1]))

File: GoLang/5 kyu/Pick peaks/main.go
package main

import "fmt"

type PosPeaks struct {
	Pos   []int
	Peaks []int
}

func main() {
	fmt.Println(PickPeaks([]int{3, 2, 3, 6, 4, 1, 2, 3, 2, 1, 2, 3}))
}
  
func goesLower(arr []int, ind int) bool {
	for k := ind; k < len(arr); k++ {
	  
	  if arr[k] > arr[ind] {
		return false
		
	  } else if arr[k] < arr[ind] {
		return true
	  }
	  
	}
	
	return false
}
  
func PickPeaks(array []int) PosPeaks {
	var pos, peaks = []int{}, []int{}
	for i := 1; i < len(array) - 1; i++ {
	  
	  if array[i] > array[i - 1] && array[i] >= array[i + 1] && goesLower(array, i) {
		pos = append(pos, i)
		peaks = append(peaks, array[i])
	  }
	  
	}
	
	result := PosPeaks {
	  Pos: pos,
	  Peaks: peaks,
	}
	
	return result
}